### ISSUE TITLE
HOW TO: weapon stats affected by particular skill

### ISSUE THREAD

**Issue by andreanespeca** [BSData/catalogue-development#78]
Basically on this game Last Days Zombie apocalypse to select a specific weapon you need to learn a specific skill so i didn't want the weapon to show up until the user select that specific skill in for that given character.
I thought i had the solution and went for hidden true if not selected that named skill from force selection, i tested and it worked than i realized that if any of the character in the roster had that skill will make the weapon available for everybody in the roster...i have tried several things which one was to unchecked the shared box but didn't work...at the moment i'm stuck and can't think at any solution to this, any help are very welcome.
Thanks

**Comment by Mad-Spy** [comment]
Hi Andrea, sounds like you need to use if equal to 0 selections of [Skill] in Parent, rather than Force.

**Comment by andreanespeca** [comment]
yeah i thought so but when i select Parent i don't get to chose much...
![image](https://user-images.githubusercontent.com/54960384/65691801-c99dd080-e071-11e9-821e-aaf01ad42fd5.png)

**Comment by Mad-Spy** [comment]
I've downloaded the repo. What's the specific situation you need to work?

**Comment by andreanespeca** [comment]
basically all the ranged weapon need to change their stats when the appropriate gunsmith (Class of the weapon) is selected for example if Gunsmith(pistol) is selected and revolver is used the stats of the revolver will change so that damage is increased by 1 and the reload is decreased by 1 if you open the repo you can see what was my solution...it did work but if anyone else is added in the roster and take the revolver stats are changed and shouldn't at least gunsmith(pistol) is selected on the new unit as well.
Also there are 3 weapon ( Squad automatic weapon/ military sniper rifle/ minigun) that are highly specialized and any gunsmith(class) need to be selected to enable them, again you can see what was my solution and works but if any other unit added will still look at the entire roster to show/hide those weapon

**Comment by andreanespeca** [comment]
i don't have discord but i can i-m installing it now

**Comment by andreanespeca** [comment]
to just make an example you should try to hide/show squad automatic weapon only if the character has the gunsmith skill (use survivor generic character to enable gunsmith skill in the list) but if another in the roster doesn't have gunsmith skill the weapon should stay hidden for him

**Comment by Mad-Spy** [comment]
Any Gunsmith skill, or one in particular?

**Comment by andreanespeca** [comment]
to show the weapon any, while to increase/decrease stats will have to be the same class of the weapon thanks

**Comment by Mad-Spy** [comment]
Right, that's what I thought.

**Comment by Mad-Spy** [comment]
Basically, it's a context issue. 

When the weapon choices and skills exist solely within their respective Shared Selection Entry Groups, they are not visible to each other. They need to exist in the Shared Selection Entries (and be linked into the SSEGs) in order to reference each other.

![image](https://user-images.githubusercontent.com/29306451/65695800-1a142e80-e070-11e9-890d-1659ecf0b39c.png)

![image](https://user-images.githubusercontent.com/29306451/65695864-344e0c80-e070-11e9-9610-9ba8f8885c01.png)

**Comment by Mad-Spy** [comment]
I've forked and done a PR for the changes.

**Comment by andreanespeca** [comment]
how do i reach you on discord?

**Comment by andreanespeca** [comment]
he works great and i see what you have done thanks for taking the time to explain that to me, it's much clearer now but the stats of the weapon don't seems to change with the gunsmith appropriate class...don't understand why

**Comment by penguin20000** [comment]
There's a discord link here https://discord.gg/UrrPB3T :)
